phasespace/fulldecay/fulldecay.py

```diff
@@ -0,0 +1,259 @@
+import itertools
+from typing import Callable, Union
+
+import tensorflow as tf
+import tensorflow.experimental.numpy as tnp
+from particle import Particle
+
+from phasespace import GenParticle
+
+from .mass_functions import _DEFAULT_CONVERTER
+
+_MASS_WIDTH_TOLERANCE = 0.01
+_DEFAULT_MASS_FUNC = "rel-BW"
+
+
+class FullDecay:
+    """A container that works like GenParticle that can handle multiple decays."""
+
+    def __init__(self, gen_particles: list[tuple[float, GenParticle]]):
+        """Create an instance of FullDecay.
+
+        Parameters
+        ----------
+        gen_particles : list[tuple[float, GenParticle]]
+            All the GenParticles and their corresponding probabilities.
+            The list must be of the format [[probability, GenParticle instance], [probability, ...
+        Notes
+        -----
+        Input format might change
+        """
+        self.gen_particles = gen_particles
+
+    @classmethod
+    def from_dict(
+        cls,
+        dec_dict: dict,
+        mass_converter: dict[str, Callable] = None,
+        tolerance: float = _MASS_WIDTH_TOLERANCE,
+    ):
+        """Create a FullDecay instance from a dict in the decaylanguage format.

```

Comment:
I realise that this is still WIP but here also I think you need to be more verbose and also add a relatively extensive examples section similarly to what I did e.g. in https://github.com/scikit-hep/particle/blob/master/src/particle/particle/particle.py#L433-L537 for `Particle.to:dict(...)` (for the sake of argument). Indeed such examples are crucial to make the user undestand how to effectively exploit the functionality, which in my example and this one method you wrote, connect 2 packages and more than one concept.

Here you need to basically explain what the dict representation of a decay chain is (a pointer to DecayLanguage cannot harm), so how to write these input dicts and pass them to this class method. The explanation of `dec_dict` could also be improved a bit.